packages/commonwealth/client/scripts/views/components/proposals/voting_results.tsx

```diff
@@ -21,14 +21,31 @@ import {
   YesNoRejectVotingResult,
 } from './voting_result_components';
 import useForceRerender from 'hooks/useForceRerender';
+import { IVote } from 'models/interfaces';
+import Proposal from 'models/Proposal';
+import useAaveProposalVotesQuery from 'state/api/proposals/aave/fetchAaveProposalVotes';
+import { ChainNetwork } from 'common-common/src/types';
+import useAaveProposalsQuery from 'state/api/proposals/aave/fetchAaveProposals';
 
 type VotingResultsProps = { proposal: AnyProposal };
 
 export const VotingResults = (props: VotingResultsProps) => {
   const { proposal } = props;
   const forceRerender = useForceRerender();
+  const [isLoading, setLoading] = useState(
+    !app.chain || !app.chain.loaded || !app.chain.apiInitialized
+  );
 
-  const votes = proposal.getVotes();
+  useEffect(() => {
```

Comment:
I have seen this effect that listens for adapter ready, being used in multiple places. I think we should extract this into a custom hook, something like this
```
export const useChainReady = () => {
  const [isLoading, setLoading] = useState(!app.chain || !app.chain.loaded);

  useEffect(() => {
    app.chainAdapterReady.on('ready', () => setLoading(false));

    return () => {
      app.chainAdapterReady.off('ready', () => setLoading(false));
    };
  }, []);

  return  {
    isLoading
  }
};
```

Reply:
It's used a lot in the Substrate components and in those there is also `app.chainModuleReady.on('ready', () => setSubstrateLoading(false));` so it can't be directly replaced by the above component. This would only be used in `proposals.tsx` and `voting_results.tsx`. Is it still worth adding?